apps/wallet/src/ui/app/pages/dapp-tx-approval/index.tsx

```diff
@@ -186,6 +208,121 @@ function TransactionSummery({ label, content, loading }: TransferSummaryProps) {
     );
 }
 
+type TransferSummerCardProps = {
+    coinSymbol: string | null;
+    amount: number | null;
+    origin: string;
+    objectId: string | null;
+    nftImage?: string | null;
+    gasEstimate: number | null;
+    isListing?: boolean;
+};
+
+function MiniNFTLink({
+    id,
+    url,
+    name,
+}: {
+    id: string;
+    url: string;
+    name?: string | null;
+}) {
+    const objectId = useMiddleEllipsis(
+        id,
+        TRUNCATE_MAX_LENGTH,
+        TRUNCATE_PREFIX_LENGTH
+    );
+
+    return (
+        <>
+            <MiniNFT url={url} name={name} />
+            <ExplorerLink
+                type={ExplorerLinkType.object}
+                objectID={id}
+                className={st.objectId}
+                showIcon={false}
+            >
+                {objectId}
+            </ExplorerLink>
+        </>
+    );
+}
+
+function TransactionSummaryCard({
+    coinSymbol,
+    amount,
+    objectId,
+    origin,
+    nftImage,
+    gasEstimate,
+    isListing,
+}: TransferSummerCardProps) {
+    const [gasEst, gasSymbol] = useFormatCoin(gasEstimate || 0, GAS_TYPE_ARG);
+
+    const [formatedAmount, symbol] = useFormatCoin(
+        amount ? Math.abs(amount) : 0,
+        coinSymbol
+    );
+
+    return (
+        <SummaryCard header="Transaction summary">
```

Comment:
I am a bit confuse isnt `Pay` type all about transactions?

Reply:
What do you mean with 'all about transactions'?

[Here](https://github.com/MystenLabs/sui/blob/e45b188a80a067700efdc5a099745f18e1f41aac/sdk/typescript/src/signers/txn-data-serializers/txn-data-serializer.ts#L27) you can see how it can be defined

Reply:
I know, but I assume this `event.coinBalanceChange?.changeType === 'Pay' && event.coinBalanceChange?.owner.AddressOwner === address` assume you own the coin.